components/lcd/esp_lcd_sh1107/README.md

```diff
@@ -0,0 +1,98 @@
+
+Packages from this repository are uploaded to [Espressif's component service](https://components.espressif.com/).
+You can add them to your project via `idf.py add-dependancy`, e.g. 
+```
+    idf.py add-dependency esp_lcd_sh1107==1.0.0
+```
+
+Alternatively, you can create `idf_component.yml`. More is in [Espressif's documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/tools/idf-component-manager.html).
+
+## Usage
+
+For detailed usage, please go to [LCD documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/lcd.html).
+
+## Example initialization 
+
+```
+#define LCD_SH1107_I2C_ADDRESS  0x3C
+#define LCD_SH1107_I2C_CMD      0X00
+#define LCD_SH1107_I2C_RAM      0X40
+
+esp_lcd_panel_io_handle_t io_handle = NULL;
+esp_lcd_panel_io_i2c_config_t io_config = {
+    .dev_addr = LCD_SH1107_I2C_ADDRESS,
+    .control_phase_bytes = 1,
+    .dc_bit_offset = 0,
+    .lcd_cmd_bits = 8,
+    .lcd_param_bits = 8,
+    .flags =
+    {
+        .disable_control_phase = 1,
+    }
+};
+ESP_ERROR_CHECK(esp_lcd_new_panel_io_i2c((esp_lcd_i2c_bus_handle_t)config->i2c.port, &io_config, &io_handle));
+
+const esp_lcd_panel_sh1107_config_t vendor_config = {
+    .lcd_width = BOARD_DISP_I2C_HRES,
+    .lcd_height = BOARD_DISP_I2C_VRES,
+};
+
+esp_lcd_panel_handle_t lcd_panel_handle = NULL;
+esp_lcd_panel_dev_config_t panel_config = {
+    .bits_per_pixel = 1,
+    .reset_gpio_num = BOARD_DISP_I2C_RST,
+    .vendor_config = (void*)&vendor_config,
+};
+ESP_ERROR_CHECK(esp_lcd_new_panel_sh1107(io_handle, &panel_config, &lcd_panel_handle));
+
+ESP_ERROR_CHECK(esp_lcd_panel_reset(lcd_panel_handle));
+ESP_ERROR_CHECK(esp_lcd_panel_init(lcd_panel_handle));
+ESP_ERROR_CHECK(esp_lcd_panel_disp_on_off(lcd_panel_handle, true));
+```
+
+## Example usage with LVGL
+
+For using with LVGL, there must be set pixel callback with specific changes of color data: 
+```
+lvgl_disp_drv->set_px_cb = disp_lvgl_pix_cb;
+```
+
+```
+static void disp_lvgl_pix_cb(lv_disp_drv_t * drv, uint8_t * buf, lv_coord_t buf_w, lv_coord_t x, lv_coord_t y,lv_color_t color, lv_opa_t opa)
+{
+    assert(drv != NULL);
+
+    /* buf_w will be ignored, the configured CONFIG_LV_DISPLAY_HEIGHT and _WIDTH,
+    and CONFIG_LV_DISPLAY_ORIENTATION_LANDSCAPE and _PORTRAIT will be used. */
+    uint16_t byte_index = 0;
+    uint8_t  bit_index = 0;
+
+    /* For landscape use these lines */
+    byte_index = y + (( x>>3 ) * drv->hor_res);
+    bit_index  = x & 0x7;
+
+    /* 
+    For portrait use these lines 
+    Note: Must be right changed flush and display size swapped:
+        call esp_lcd_panel_swap_xy(panel, true) 
+        and esp_lcd_panel_mirror(panel, false/true, false/true) if needed
+    */
+    //byte_index = x + (( y>>3 ) * drv->ver_res);
+    //bit_index  = y & 0x7;
+
+    if (lv_color_to1(color)) {
+        BIT_SET(buf[byte_index], bit_index);
+    } else {
+        BIT_CLEAR(buf[byte_index], bit_index);
+    }
+}
```

Comment:
We should discuss how to distribute this function in a better way. I don't like leaving it in README.md. But no idea right now... :(

Reply:
I understand, I looked for the better solution, but not found. I will think about it.